From b77e1513e8c5feb3d69790be7651c6889e045dc4 Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Mon, 9 Dec 2024 15:10:25 +0800
Subject: [PATCH] Arch Linux fix LTO

---
 src/libpcp/src/check-statics | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libpcp/src/check-statics b/src/libpcp/src/check-statics
index 29dcb868a..3bc3ed808 100755
--- a/src/libpcp/src/check-statics
+++ b/src/libpcp/src/check-statics
@@ -721,6 +721,7 @@ do
 		    -e 's/b \([_a-zA-Z][_a-zA-Z0-9]*\)\.[0-9]*.[0b]$/b \1/' \
 		    -e 's/\([bds] \).*\.\([_a-zA-Z]\)/\1\2/' \
 		    -e 's/s _glib_relative_date\./s /' \
+		    -e '/ B __gnu_lto_slim/d' \
 		    -e '/ s EH_/d' \
 		    -e '/ s str\.[0-9][0-9]*$/d' \
 		    -e '/ b \.bss/d' \
-- 
2.34.1

