From 5d8cd45ed5de01ce298ecba63d0da32e4c83d221 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Wed, 4 Dec 2024 02:20:58 +0800
Subject: [PATCH 129/132] AOSCOS: DEBUG

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 kernel/irq/manage.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index f0803d6bd296..4ae4bf68b343 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -2174,20 +2174,24 @@ int request_threaded_irq(unsigned int irq, irq_handler_t handler,
 	    (!(irqflags & IRQF_SHARED) && (irqflags & IRQF_COND_SUSPEND)) ||
 	    ((irqflags & IRQF_NO_SUSPEND) && (irqflags & IRQF_COND_SUSPEND)))
 		return -EINVAL;
+	pr_err("1");
 
 	desc = irq_to_desc(irq);
 	if (!desc)
 		return -EINVAL;
+	pr_err("2");
 
 	if (!irq_settings_can_request(desc) ||
 	    WARN_ON(irq_settings_is_per_cpu_devid(desc)))
 		return -EINVAL;
+	pr_err("3");
 
 	if (!handler) {
 		if (!thread_fn)
 			return -EINVAL;
 		handler = irq_default_primary_handler;
 	}
+	pr_err("4");
 
 	action = kzalloc(sizeof(struct irqaction), GFP_KERNEL);
 	if (!action)
-- 
2.47.1

