From 17342f09eebb1890b7f818e10000e272b49775ef Mon Sep 17 00:00:00 2001
From: Huacai Chen <chenhc@lemote.com>
Date: Thu, 1 Feb 2018 14:06:27 +0800
Subject: [PATCH 217/273] AOSCOS: writeback: Limit maximum dirty data for
 Loongson-3

Signed-off-by: Huacai Chen <chenhc@lemote.com>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 mm/page-writeback.c | 33 +++++++++++++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/mm/page-writeback.c b/mm/page-writeback.c
index d213ead95675..4c9e1a15cfa3 100644
--- a/mm/page-writeback.c
+++ b/mm/page-writeback.c
@@ -29,6 +29,7 @@
 #include <linux/mpage.h>
 #include <linux/rmap.h>
 #include <linux/percpu.h>
+#include <linux/sizes.h>
 #include <linux/smp.h>
 #include <linux/sysctl.h>
 #include <linux/cpu.h>
@@ -2402,6 +2403,38 @@ static struct ctl_table vm_page_writeback_sysctls[] = {
  */
 void __init page_writeback_init(void)
 {
+#ifdef CONFIG_CPU_LOONGSON3
+	uint64_t x, d;
+	unsigned long ratio = dirty_background_ratio;
+	unsigned long avail = global_dirtyable_memory();
+	unsigned long limit = 2*800*1024*1024UL / PAGE_SIZE;
+
+	if (avail*ratio > limit*100) {
+		dirty_background_ratio = 0;
+		vm_dirty_ratio = 0;
+		dirty_background_bytes = limit * PAGE_SIZE;
+		vm_dirty_bytes = 2 * dirty_background_bytes;
+	}
+	dirty_expire_interval = 10 * 100;
+
+	avail = avail * PAGE_SIZE / SZ_1G;
+
+	switch (avail) {
+	case 0 ... 89:
+		d = avail / 2;
+		break;
+	case 90 ... 269:
+		x = avail / 4;
+		x = x*x*x*x*x*x*x*x*x*x;
+		d = fls64(x) - 1;
+		break;
+	default:
+		/* 270 ... ULONG_MAX */
+		d = vm_swappiness;
+	}
+	vm_swappiness -= d;
+#endif
+
 	BUG_ON(wb_domain_init(&global_wb_domain, GFP_KERNEL));
 
 	cpuhp_setup_state(CPUHP_AP_ONLINE_DYN, "mm/writeback:online",
-- 
2.47.1

